### ISSUE TITLE
argh, a bug in caching of some sort!

### ISSUE THREAD

**Issue by ishanshrivastava2011** [JohnLangford/vowpal_wabbit#1261]
# **This is the code I ran.**
vw house_dataset -l 10 -c --passes 25 --holdout_off

## **This is how the house_dataset file looks:**
0 |u price:.23 sqft:.25 |v age:.05 2006
1 2 'second_house |u price:.18 sqft:.15 |v age:.35 1976
0 1 0.5 'third_house |u price:.53 sqft:.32 |v age:.87 1924

## **Running that code gives the following output:**
Num weight bits = 18
learning rate = 10
initial_t = 0
power_t = 0.5
decay_learning_rate = 1
creating cache_file = house_dataset.cache
Reading datafile = house_dataset
num sources = 1
average  since         example        example  current  current  current
loss     last          counter         weight    label  predict features
0.000000 0.000000            1            1.0   0.0000   0.0000        5
0.666667 1.000000            2            3.0   1.0000   0.0000        5
error, failed to write example
vw example #3(parser.cc:272): argh, a bug in caching of some sort!

## **Adding -k to the command also did not help. This is the solution that is given here :** https://github.com/JohnLangford/vowpal_wabbit/issues/1247

Please help. I remember this command running properly somewhere till mid of May 2017. 
Don't understand why this has started popping up these days.

**Comment by arielf** [comment]
Thanks for the report.

This seems very weird and localized.

I can't reproduce this with latest `vw` on Ubuntu 16.04.2:

```
$ cat house_dataset
0 |u price:.23 sqft:.25 |v age:.05 2006
1 2 'second_house |u price:.18 sqft:.15 |v age:.35 1976
0 1 0.5 'third_house |u price:.53 sqft:.32 |v age:.87 1924

$ vw house_dataset  -l 10 -c --passes 25 --holdout_off -k
Num weight bits = 18
learning rate = 10
initial_t = 0
power_t = 0.5
decay_learning_rate = 1
creating cache_file = house_dataset.cache
Reading datafile = house_dataset
num sources = 1
average  since         example        example  current  current  current
loss     last          counter         weight    label  predict features
0.000000 0.000000            1            1.0   0.0000   0.0000        5
0.666667 1.000000            2            3.0   1.0000   0.0000        5
0.589385 0.531424            5            7.0   1.0000   0.2508        5
0.378923 0.194769           11           15.0   1.0000   0.8308        5
0.184476 0.002182           23           31.0   1.0000   0.9975        5
0.090774 0.000000           47           63.0   1.0000   1.0000        5

finished run
number of examples per pass = 3
passes used = 25
weighted example sum = 100.000000
weighted label sum = 50.000000
average loss = 0.057188
best constant = 0.500000
best constant's loss = 0.250000
total feature number = 375
```

Failing to write example into cache file after processing 2 lines successfully is surprising.

Do you have permissions to write in current dir?
If yes, Is you hardware ok?  Any errors in the system logs?

What kind of environment this is running on? Which version of `vw`?

**Comment by ishanshrivastava2011** [comment]
Hi, 
I guess there was some issue with the VM, I was using.
Considering your reply, I thought switching to a different VM, and it all worked just fine.
I guess we can close this issue then.

Thanks

**Comment by MLWave** [comment]
Referenced #1247 

You should be able to close your own issue.

**Comment by JohnLangford** [comment]
Thanks Ariel
